### PR TITLE
fix(cache): fix dedup error of meta-cache

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/storage/src/hummock/sstable_store.rs [hunk 1]
```diff
@@ -330,38 +330,38 @@ impl SstableStore {
...
+                    let store = self.store.clone();
+                    let meta_path = self.get_sst_data_path(sst_id);
+                    stats.cache_meta_block_miss += 1;
+                    let stats_ptr = stats.remote_io_time.clone();
+                    let loc = BlockLocation {
+                        offset: sst.meta_offset as usize,
+                        size: (sst.file_size - sst.meta_offset) as usize,
+                    };
+                    async move {
+                        let now = Instant::now();
```
Comment (Li0k) [inline src/storage/src/hummock/sstable_store.rs:346]: we can use `minstant::Instant`

**Comment by BugenZhao** [comment]
Should we wrap this logic in `LruCache`?

File: src/storage/src/hummock/sstable_store.rs [hunk 2]
```diff
@@ -330,38 +330,38 @@ impl SstableStore {
...
-            })
-            .verbose_stack_trace("meta_cache_lookup")
-            .await
-            .map_err(|e| {
-                HummockError::other(format!(
-                    "meta cache lookup request dedup get cancel: {:?}",
-                    e,
-                ))
-            })?
+        loop {
```
Comment (Li0k) [inline src/storage/src/hummock/sstable_store.rs:333]: How to find out if an infinite loop occurs?

**Comment by BugenZhao** [comment]
> Should we encapsulate this logic in `LruCache`?

For example, `moka` will automatically retry if the future aborts.

```rust
                        // Somebody else (a future containing `get_with`/`try_get_with`)
                        // has been aborted.
                        WaiterValue::EnclosingFutureAborted => {
                            retries += 1;
                            panic_if_retry_exhausted_for_aborting(retries, MAX_RETRIES);
                            // Retry from the beginning.
                            continue;
                        }
```

**Review by hzxa21 — APPROVED** [review]

**Comment by hzxa21** [comment]
> > Should we encapsulate this logic in `LruCache`?
> 
> For example, `moka` will automatically retry if the future aborts.
> 
> ```rust
>                         // Somebody else (a future containing `get_with`/`try_get_with`)
>                         // has been aborted.
>                         WaiterValue::EnclosingFutureAborted => {
>                             retries += 1;
>                             panic_if_retry_exhausted_for_aborting(retries, MAX_RETRIES);
>                             // Retry from the beginning.
>                             continue;
>                         }
> ```

Good point. It is better to put the logic in the implementation of `lookup_with_request_dedup`.

**Comment by Little-Wallace** [comment]
> Should we encapsulate this logic in `LruCache`?

Done

**Review by BugenZhao — APPROVED** [review]

**Review by Li0k — APPROVED** [review]

**Pull request merged**